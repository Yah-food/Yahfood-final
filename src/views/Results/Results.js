import React from 'react';
import RecipeCard from '../../components/RecipeCard/RecipeCard';
import { Link } from 'react-router-dom';
export default function Results({ currentResults }) {
  return (
    <>
      <div>
        <div className="flex-row bg-lime-700/50 h-24 rounded-b-3xl mb-5 shadow-lg">
          <Link to="/profile" className="mx-5 py-5">
            <button>Back</button>
          </Link>
          <div className="flex justify-center font-sans text-5xl opacity-70 pt-2">
            <h1>yahfood!</h1>
          </div>
        </div>
      </div>
      <div className="flex pt-48 overflow-x-auto">
        <RecipeCard allRecipes={currentResults} />
      </div>
    </>
  );
}
