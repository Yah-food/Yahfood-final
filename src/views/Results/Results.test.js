import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Results from './Results';

test('renders the recipe card on the results page', async () => {
  const currentResults = [
    {
      id: 5,
      title: 'Fried Chicken',
      images:
        'https://www.tasteofhome.com/wp-content/uploads/2018/01/Best-Ever-Fried-Chicken_EXPS_MCMZ16_23240_B07_12_3b.jpg?resize=700,700',
    },
  ];
  const { container } = render(
    <MemoryRouter>
      <Results currentResults={currentResults} />
    </MemoryRouter>
  );
  await expect(screen.findByText('Fried Chicken'));
  expect(container).toMatchSnapshot();
});
