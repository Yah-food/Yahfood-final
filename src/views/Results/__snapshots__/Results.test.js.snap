// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the recipe card on the results page 1`] = `
<div>
  <div>
    <div>
      <div
        class="flex flex-row overflow-x-auto"
      >
        <div>
          <h3
            class="flex justify-center"
          >
            Fried Chicken
          </h3>
          <a
            href="/recipe/Fried Chicken"
          >
            <div
              class="object-cover h-48 w-48 flex justify-center m-5"
            >
              <img
                src="https://www.tasteofhome.com/wp-content/uploads/2018/01/Best-Ever-Fried-Chicken_EXPS_MCMZ16_23240_B07_12_3b.jpg?resize=700,700"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
